app/controllers/spree/orders_controller.rb

```diff
@@ -69,7 +69,7 @@ def populate
         format.html do
           if @order.errors.any?
             flash[:error] = @order.errors.full_messages.join(", ")
-            redirect_back_or_default(spree.root_path)
+            redirect_back(fallback_location: spree.root_path)
```

Comment:
I see that `redirect_back` is only soft deprecated, so let's do that. Ideally, we should remember to update to `redirect_back_or_to` once we remove support for Rails 6.